### PR TITLE
feat(GuildBanManager): Support pagination results

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
See https://github.com/discord/discord-api-docs/commit/808d94b82c04d734b622d23688a3cdbe2e0903ea.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/discord.js/src/managers/GuildBanManager.js [hunk 1]
```diff
@@ -55,22 +55,25 @@ class GuildBanManager extends CachedManager {
...
+   * @property {boolean} cache Whether to cache the fetched bans
    */
 
   /**
    * Fetches ban(s) from Discord.
    * @param {UserResolvable|FetchBanOptions|FetchBansOptions} [options] Options for fetching guild ban(s)
    * @returns {Promise<GuildBan|Collection<Snowflake, GuildBan>>}
    * @example
-   * // Fetch all bans from a guild
+   * // Fetch multple bans from a guild
```
Comment (DTrombett) [inline packages/discord.js/src/managers/GuildBanManager.js:71]:
```suggestion
   * // Fetch multiple bans from a guild
```

**Review by DTrombett — APPROVED** [review]

File: packages/discord.js/src/managers/GuildBanManager.js [hunk 2]
```diff
@@ -112,8 +116,16 @@ class GuildBanManager extends CachedManager {
...
-    const data = await this.client.rest.get(Routes.guildBans(this.guild.id));
+  async _fetchMany({ cache, limit, before, after } = {}) {
+    const options = {};
+    if (limit) options.limit = limit;
+    if (before) options.before = before;
+    if (after) options.after = after;
+
+    const data = await this.client.rest.get(Routes.guildBans(this.guild.id), {
+      query: new URLSearchParams(options),
+    });
```
Comment (suneettipirneni) [inline packages/discord.js/src/managers/GuildBanManager.js:127]: You can just reuse the passed object here.

```suggestion
  async _fetchMany(options) {
    const data = await this.client.rest.get(Routes.guildBans(this.guild.id), {
      query: new URLSearchParams(options ?? {}),
    });
```

**Review by suneettipirneni — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/managers/GuildBanManager.js (hunk 2 above)
Comment (Jiralite) [reply packages/discord.js/src/managers/GuildBanManager.js:127]: Performing `<GuildBanManager>.fetch({ limit: 5 })` with your suggestion will cause Discord to reject the request as the other values are being explicitly sent as `undefined`.

Comment (suneettipirneni) [reply packages/discord.js/src/managers/GuildBanManager.js:127]: It wouldn't, in the code snippet you gave you never set the other missing props as `undefined` so they don't get serialized at all. See [this playground](https://www.typescriptlang.org/play?#code/FAFwngDgpgBA8hEBLA9gOwM4wLwwN7AxEwBmKKA-AFwwYgBOSaA5gNyHEBGAhvdbQyZtgAX2DASAVzQBjZOhggodABQpEqTDQTzMASnwciM9BhQAbKADpzKZirRQA7jACqAJQAyAZSi8ZABYACrzcALYYahqmelYgKN6CLCp6eqLiSqoExKTkNADk6FD5onpAA)

Comment (kyranet) [reply packages/discord.js/src/managers/GuildBanManager.js]: Things fail when you add an undefined key. See [this other playground](https://www.typescriptlang.org/play?#code/FAFwngDgpgBA8hEBLA9gOwM4wLwwN7AxEwBmKKA-AFwwYgBOSaA5gNyHEBGAhvdbQyZtgAX2DASAVzQBjZOhggodABQpEqTDQTzMASnwciM9BhQAbKADpzKZirRQA7jACqAJQAyAZSi8ZABYACrzcALYYahqmelYgKN6CLCp6eqLiSqoExKTkNADk6FD5ADQwPPQ00gAmUCRMUNWiekA).

**Review by RealAlphabet — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**